services/directions.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+var v = require('./service-helpers/validator');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+
+/**
+ * Directions API service.
+ */
+var Directions = {};
+
+/**
+ * Get directions.
+ *
+ * See the [Mapbox Directions API](https://www.mapbox.com/api-documentation/#directions).
+ *
+ * @param {Object} config
+ * @param {'driving-traffic'|'driving'|'walking'|'cycling'} [config.profile=driving]
+ * @param {Array.<Object>} config.way_points - An ordered array of objects with `latitude`
```

Comment:
I think it would be very helpful as a user to explain how I can look up the docs about each of the options in these objects. Maybe just a sentence about how these are listed as independent query strings whose length must match the waypoints' length, but they are combined here into one object for each waypoint.

Reply:
As I was unable to add options properties, I have moved the entire `wayPoints` type into a typeDef.